src/objects.c

```diff
@@ -387,6 +387,42 @@ PgUser *add_user(const char *name, const char *passwd)
 	return user;
 }
 
+/* find and use original or preconfigured user given the login user */
+void use_original_user(PgUser **user)
+{
+	/*
+	 * The original user will either be the current login user, a preconfigured user via
+	 * ini file [users] section, a preconfigured user via global auth_file, a PAM user,
+	 * a user attached to a database via auth_query, a forced user via user= entry via
+	 * ini file [databases] section, or not exist yet if the user has never connected.
+	 */
+	PgUser *login_user = *user;
+	PgUser *original_user = find_user(login_user->name);
+
+	/* add new login user to reload their configurations at runtime */
+	if (original_user == NULL) {
```

Comment:
If the new login user is not defined in ini configuration `[users]`/`[databases]` section, global `auth_file`, etc. then we store it in the global `user_list` to ensure that any admin `RELOAD`s can access the user to update it's configuration fields at runtime.